.orchestra/ci/ci-hook/app.py

```diff
@@ -44,11 +53,54 @@ def get_env_or_fail(var_name):
 mapping = config["github_to_gitlab_mapping"]
 default_user = config["default_user"]
 ci_user = config["ci_user"]
+github_app_id = config["github_app_id"]
+ci_job_url = config["ci_job_url"]
+github_installation_id = config["github_installation_id"]
+
+_installation_token_info = None
+
+
+def installation_token() -> str:
+    global _installation_token_info
+    should_refresh = _installation_token_info is None
+    if not should_refresh:
+        expires = dateutil.parser.isoparse(_installation_token_info["expires_at"])
+        now = datetime.now(expires.tzinfo)
+        should_refresh = expires - timedelta(minutes=1) > now
+
+    if should_refresh:
+        token = jwt.encode(
+            {
+                "iat": int(time.time()) + 60,
+                "exp": int(time.time()) + 10 * 60,
+                "iss": str(github_app_id)
+            },
+            github_priv_key,
+            "RS256"
+        )
+        r = requests.post(f"{GITHUB_API_URL}/app/installations/{github_installation_id}/access_tokens", headers={
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {token}",
+        })
+        if r.status_code not in (200, 201, 202):
+            raise RuntimeError("Unable to retrieve GitHub installation token")
+        _installation_token_info = r.json()
+    return _installation_token_info["token"]
+
+
+def github_headers(headers: Optional[dict] = None) -> dict:
```

Comment:
Unused arguments are a bit sketchy to me. We could introduce them when needed.
But no biggie.